### ISSUE TITLE
kustomize panic

### ISSUE THREAD

**Issue by Liujingfang1** [kubernetes-sigs/kustomize#808]
```
panic: interface conversion: interface {} is map[string]interface {}, not []interface {}

goroutine 1 [running]:
sigs.k8s.io/kustomize/pkg/transformers.(*imageTransformer).updateContainers(0xc00025cf20, 0xc0004e3710, 0x100b88b, 0xa, 0xc0003a6e08, 0x60000c0003a6301)
    /home/don/go/src/sigs.k8s.io/kustomize/pkg/transformers/image.go:78 +0x4fc
sigs.k8s.io/kustomize/pkg/transformers.(*imageTransformer).findAndReplaceImage(0xc00025cf20, 0xc0004e3710, 0xc000364df0, 0x4)
    /home/don/go/src/sigs.k8s.io/kustomize/pkg/transformers/image.go:65 +0x13e
sigs.k8s.io/kustomize/pkg/transformers.(*imageTransformer).findContainers(0xc00025cf20, 0xc0004e3650, 0x100e547, 0xe)
    /home/don/go/src/sigs.k8s.io/kustomize/pkg/transformers/image.go:112 +0x1f2
sigs.k8s.io/kustomize/pkg/transformers.(*imageTransformer).findAndReplaceImage(0xc00025cf20, 0xc0004e3650, 0xc000364d38, 0x8)
    /home/don/go/src/sigs.k8s.io/kustomize/pkg/transformers/image.go:72 +0xe1
sigs.k8s.io/kustomize/pkg/transformers.(*imageTransformer).findContainers(0xc00025cf20, 0xc0004e35c0, 0x100e547, 0xe)
    /home/don/go/src/sigs.k8s.io/kustomize/pkg/transformers/image.go:112 +0x1f2
sigs.k8s.io/kustomize/pkg/transformers.(*imageTransformer).findAndReplaceImage(0xc00025cf20, 0xc0004e35c0, 0xc0003652dc, 0x4)
    /home/don/go/src/sigs.k8s.io/kustomize/pkg/transformers/image.go:72 +0xe1
sigs.k8s.io/kustomize/pkg/transformers.(*imageTransformer).findContainers(0xc00025cf20, 0xc0004e3560, 0x100e547, 0xe)
    /home/don/go/src/sigs.k8s.io/kustomize/pkg/transformers/image.go:112 +0x1f2
sigs.k8s.io/kustomize/pkg/transformers.(*imageTransformer).findAndReplaceImage(0xc00025cf20, 0xc0004e3560, 0x0, 0x0)
    /home/don/go/src/sigs.k8s.io/kustomize/pkg/transformers/image.go:72 +0xe1
sigs.k8s.io/kustomize/pkg/transformers.(*imageTransformer).Transform(0xc00025cf20, 0xc0003eca80, 0x0, 0x0)
    /home/don/go/src/sigs.k8s.io/kustomize/pkg/transformers/image.go:45 +0xca
sigs.k8s.io/kustomize/pkg/transformers.(*multiTransformer).transform(0xc0003ed620, 0xc0003eca80, 0x0, 0xc0003641f0)
    /home/don/go/src/sigs.k8s.io/kustomize/pkg/transformers/multitransformer.go:62 +0x6f
sigs.k8s.io/kustomize/pkg/transformers.(*multiTransformer).Transform(0xc0003ed620, 0xc0003eca80, 0xc0003eca80, 0xc0004e16c0)
    /home/don/go/src/sigs.k8s.io/kustomize/pkg/transformers/multitransformer.go:58 +0x75
sigs.k8s.io/kustomize/pkg/target.(*ResAccumulator).Transform(0xc000030ea0, 0x1136fa0, 0xc0003ed620, 0x0, 0xc0001288f0)
    /home/don/go/src/sigs.k8s.io/kustomize/pkg/target/resaccumulator.go:121 +0x3e
sigs.k8s.io/kustomize/pkg/target.(*KustTarget).AccumulateTarget(0xc000030e40, 0xc00030b540, 0xc00000e728, 0x113a680)
/home/don/go/src/sigs.k8s.io/kustomize/pkg/target/kusttarget.go:209 +0x3b3
sigs.k8s.io/kustomize/pkg/target.(*KustTarget).accumulateBases(0xc00019fdd0, 0xc00035adc0, 0x30)
```

**Comment by narg95** [comment]
I took a look and as far as I cann see the only reason for it to happen is when the `initContainers` or `containers` path is not an `array` but a `map`. Looking at the kubernetes specification https://kubernetes.io/docs/reference/ I can not find that `map` is a valid use case. So the only solution that I see is to prevent the panic and instead to provide a meaningful error message.

@Liujingfang1 what is your opinion? and can you provide the steps to reproduce it?

**Comment by Liujingfang1** [comment]
For the goroutine, it's from image transformer. I believe this has been fixed by #809.
